dev/src/bundle.ts

```diff
@@ -168,6 +159,18 @@ export class BundleBuilder {
         ]);
       }
     }
+
+    const metadata: firestore.IBundleMetadata = {
+      id: this.bundleId,
+      createTime: this.latestReadTime.toProto().timestampValue,
+      version: BUNDLE_VERSION,
+      totalDocuments: this.documents.size,
+      totalBytes: bundleBuffer.length,
+    };
+    bundleBuffer = Buffer.concat([
+      this.elementToLengthPrefixedBuffer({metadata}),
+      bundleBuffer,
+    ]);
```

Comment:
I am not sure I understand how this will work. I assume this was added to allow the progress handler to display their status. For that to work, would this not need to be the first message?

Reply:
Note the order of the arguments to `Buffer.concat` is different here. Also added a comment.